scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -354,14 +367,31 @@
    the offer.
 
    Returns a list of tasks that got matched to the offer"
-  [schedule {{:keys [cpus mem]} :resources :as offer}]
-  (log/debug "Matching offer " offer " to the schedule" schedule)
-  (loop [[candidates & remaining-prefixes] (prefixes schedule)
-         prev []]
-    (let [required (util/sum-resources-of-jobs candidates)]
-      (if (and candidates (>= cpus (:cpus required)) (>= mem (:mem required)))
-        (recur remaining-prefixes candidates)
-        prev))))
+  [^TaskScheduler fenzo considerable offers db fid]
+  (log/debug "Matching" (count offers) "offers to" (count considerable) "jobs with fenzo")
+  (let [t (System/currentTimeMillis)
+        leases (mapv #(->VirtualMachineLeaseAdapter % t) offers)
+        requests (mapv (fn [job]
+                         (let [job-id (:db/id job)]
+                           (->TaskRequestAdapter job (job->task-info db fid (:db/id job)))))
+                       considerable)
+        result (.scheduleOnce fenzo requests leases)
+        failure-results (.. result getFailures values)
+        assignments (.. result getResultMap values)]
+    (log/info "Found this assigment:" result)
```

Comment:
We should also look at and log these: http://netflix.github.io/Fenzo/fenzo-core/com/netflix/fenzo/TaskScheduler.html#scheduleOnce-java.util.List-java.util.List-